### ISSUE TITLE
[SVG] `multline` width too large inside HTML `<details>`

### ISSUE THREAD

**Issue by hbghlyj** [mathjax/MathJax#2990]
### Steps to Reproduce:

1. [MathJax demo](https://www.mathjax.org/#demo)
2. In context menu `Math Settings > Math Renderer` select SVG
3. Write `multline` environment inside HTML `<details>`
```tex
<details>
\begin{multline*}1\\1\end{multline*}
</details>
```
The second line is far out. (watch the horizontal scrollbar)
4. Inspect the svg element. It is of width 89552.372ex !
![image](https://user-images.githubusercontent.com/53823634/213866732-c01c6973-666f-4968-a778-d905c5e14206.png)

### Technical details:

* MathJax Version: 3
* Client OS: Windows 11
* Browser: Firefox 108.0.1

**Comment by dpvc** [comment]
Because the contents of `<details>` effectively have `display:none` when it is closed, MathJax can't measure the available width for the math contained within them.  In that situation, MathJax considers the width to be "infinite" (i.e., a very large number).  Because the `multline` environment makes a table that is 100% width, that makes it very wide.

One work around would be to add `details` to the skipped HTML tag list and add event listeners to the details elements so that when they open for the first time, they run MathJax on their contents.  That way, the contents will not be `display:none`, and so MathJax will get the correct width.

Here is an example that does this:

``` html
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8" />
<meta content="width=device-width, initial-scale=1" name="viewport"/>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<title>Managing details elements</title>
<script>
MathJax = {
  options: {
    skipHtmlTags: {'[+]': ['details']}
  }
};
</script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js" defer></script>
</head>
<body>

<details>
\begin{multline*}1\\1\end{multline*}
</details>

<script>
function toggleDetails(event) {
  const details = event.target;
  if (details.hasAttribute("open")) {
    MathJax.startup.promise = MathJax.startup.promise.then(
      () =>  MathJax.typesetPromise(details.childNodes)
    );
    details.removeEventListener('toggle', toggleDetails);
  }
}

for (const details of document.querySelectorAll('details')) {
  details.addEventListener('toggle', toggleDetails);
}
</script>
</body>
</html>
```

**Comment by dpvc** [comment]
Here is an alternative that prevents the un-typeset mathematics from showing before it is typeset.  It does flicker the `details` summary text, but at least it doesn't show the LaTeX code.

``` html
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="utf-8" />
<meta content="width=device-width, initial-scale=1" name="viewport"/>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<title>Managing details elements</title>
<script>
MathJax = {
  options: {
    skipHtmlTags: {'[+]': ['details']}
  }
};
</script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js" defer></script>
<style>
details[open].pre-mathjax {
  visibility: hidden;
}
</style>
</head>
<body>

<details>
\begin{multline*}1\\1\end{multline*}
</details>

<script>
function toggleDetails(event) {
  const details = event.target;
  if (details.hasAttribute("open")) {
    MathJax.startup.promise = MathJax.startup.promise.then(
      () => MathJax.typesetPromise(details.childNodes)
    ).then(() => details.classList.remove('pre-mathjax'));
    details.removeEventListener('toggle', toggleDetails);
  }
}

for (const details of document.querySelectorAll('details')) {
  details.addEventListener('toggle', toggleDetails);
  details.classList.add('pre-mathjax');
}
</script>
</body>
</html>
```

**Comment by lucasvreis** [comment]
I came across this issue, and while the suggestion for rendering the math on display works for me, it brings the issue of state across equations: tag numbers may get out of order, cross references and macros may not be defined yet, etc. So I was wondering whether there are methods to process math fragments without rendering them (just for state's sake), and delay the render phase until they are visible. If it's not possible, I kindly propose something like this as an useful addition to the API.

As always @dpvc thank you for your work in MathJax.

**Comment by dpvc** [comment]
@lucasvreis, I'm not entirely sure how numbering should be handled with content that isn't showing initially.  It seems finny to me to see equation 1 and 3, where equation 2 is hidden inside a `details` element.

But here is a script you can use to get everything typesetting initially, but re-render the details when they are first opened to get the size right for `multline` elements.  All the math actually is typeset and inserted originally (you can't really do only part of the process without a lot more work), but then when the detail is opening, we set the contents of the details container to think they are back to the state that is after compiling the TeX but before getting the surrounding metric data and re-typeset from there.  For this, you *don't* want to set the `skipHtmlTags`, or the details will not get processed initially, which is the problem with the previous code.

``` html
<script>
function toggleDetails(event) {
  const details = event.target;
  if (details.hasAttribute("open")) {
    const {STATE} = MathJax._.core.MathItem;
    MathJax.startup.promise = MathJax.startup.promise.then(() => {
      for (const item of MathJax.startup.document.getMathItemsWithin(details)) {
        item.state(STATE.CONVERT + 1);
      }
      return MathJax.typesetPromise(details.childNodes).then(
        () => details.classList.remove('pre-mathjax')
      );
    });
    details.removeEventListener('toggle', toggleDetails);
  }
}

for (const details of document.querySelectorAll('details')) {
  details.addEventListener('toggle', toggleDetails);
  details.classList.add('pre-mathjax');
}
</script>
```

The rest of the file stays the same, other than removing the `skipHtmlTags`.
